### PR TITLE
adds option to hide virtual rides from activity feed

### PR DESCRIPTION
**Pull request by benoj**
Hi,

This PR adds the option to hide virtual rides from the activity feed (I have noticed a lot of them on my feed and I would like the ability to hide them).

It also resolves activity type resolution in the activity feed hider module. Looks like the html structure has changed from when the initial code was written which I guess would mean the minimum distance ride/run filters may not have been working, unless I have misinterpreted what the original code was doing.

This is my first typescript PR so let me know if it is not stylistically correct. I have tried to follow the established patterns in the code as closely as possible.

I think there could be room for improvement around working out whether or not the modifiers should run or not? Maybe using a strategy pattern or something? I didn't want to refactor too much as I am not sure how open to that you are?

Thanks,

Ben

### CODE REVIEW

**Comment by thomaschampagne** [comment]
@benoj Very thanks for this work & involvement :) :) !! I wanted to do that also with "Hide commutes" option. Great :)

Unfortunately i can't get your PR working :( And lost "Hide ride under X"

Thanks :) :)

**Comment by benoj** [comment]
@thomaschampagne I am happy to do the "Hide Commutes" option too if you like? 

Sorry just realised that I deleted the ride under! that wasn/t intentional . I have resolved that now. When you say you can't get it working, what do you mean? It won't compile or wont work? seems to be working when I run on my laptop

**Comment by thomaschampagne** [comment]
@benoj yes could be cool 👍 It's hard to me to concentrate on this with heavy work on other big feature.

i can compile. I mean i can see virtual rides with option enabled :(

Thanks for the work man ! Really appreciated !

Tom

**Comment by benoj** [comment]
Ah yes, can you try clearing your local storage. I had this same issue, looks like its to do with adding new user settings. The code which loads the user settings seems to ignore loading IF there is already something in the chrome storage meaning that adding new settings requires a clear :(

**Comment by benoj** [comment]
@thomaschampagne  I have added a new check on the load of the settings, which will check if the keys are the same between default settings and chrome synced settings. If it has changed (new setting added/removed) then it adds the key to the chrome synced settings

**Comment by thomaschampagne** [comment]
Perfect works like a charm :) 
Thanks !

File: plugin/core/scripts/Content.ts [hunk 1]
```diff
@@ -74,6 +74,12 @@ class Content {
                         chromeSettings = this.userSettings;
                     }
 
+                    let defaultSettings = _.keys(userSettings)
+                    let syncedSettings = _.keys(chromeSettings)
+                    if(_.difference(defaultSettings, syncedSettings).length !== 0){ // If settings shape has changed
+                       _.defaults(chromeSettings, userSettings)
```
Comment (thomaschampagne) [inline plugin/core/scripts/Content.ts]: Missing affectation no? 
`let XXX = _.defaults(chromeSettings, userSettings)`

>@Returns: `object` with added `defaults` values.

Comment (benoj) [inline plugin/core/scripts/Content.ts]: Underscore defaults assignes to the object as opposed to creating a new object like Object.assign. http://underscorejs.org/#defaults unless i have misread the docs.. but seems to be working

Comment (thomaschampagne) [inline plugin/core/scripts/Content.ts]: ok :)

**Pull request merged**